Yg dev branch

ygSonggoon opened the pull request:
사전에 저희가 정의 했던 약속들을 적용한 파일들 입니다. 해당 내용을 한글로 번역하고 주석처리만 추가로 작업을 했습니다.

해당 내용에서 주 검사 대상은 매크로명과 함수별 이름이 될 것 같습니다.
확인 부탁드립니다.

ygSonggoon committed:
[fix, DOC] - [에러 발생시 진행되어야 하는 부분]

에러가 발생할 때 출력이 없다면,
코드를 수정하는데 오래 걸릴 것이라고 판단했습니다.
따라서 첫번째 조건문[배열 범위를 벗어나는지 검사]에서 종료전 출력할 수 있도록 조치를 취했습니다.

ygSonggoon committed:
[add] - [방향키 조정기능]

약속된 변수명을 사용해서 별도로 파일로 작성을 했습니다.
특이사항은 없습니다.

ygSonggoon committed:
Merge branch 'master' into ygDevBranch

ygSonggoon committed:
슬랙 연동확인을 위한 테스트

ygSonggoon committed:
Slack 연동을 위한 테스트

ygSonggoon committed:
for test

Merge branch 'master' into ygDevBranch

ygSonggoon committed:
[add, fix] - [맡은 부분 주석처리 및 각 기능을 각 파일로 분할하기]

스페인어로 되어있던 코드를 전부 한국어/영어로 바꾼 후
각 코드 의미에 맞게 주석처리를 하였습니다.
아래 내용은 각 함수명과 기능에 대해 서술해 놓은 자료 입니다.

chage_direction.c :
 팩맨의 방향 상태를 바꾸어주는 기능을 합니다. 상태를 저장하기 위해서 구조체 변수를 이용합니다.
 게임 시작 전에는 정지 상태에 있다가, 값이 들어오면 상태를 변경합니다.

collision_wall_check.c :
 이동 할 위치가 벽인지 사전에 판단해주는 기능을 합니다.
 우선 지도는 배열에 저장되므로 배열의 인덱스가 음수이거나 범위를 넘어설 경우는 에러처리를 하고,
 아닌 경우 해당 배열의 값을 검사해서 벽인지 아닌지 판별해 냅니다.

move_pacman.c :
 팩맨의 위치를 변경하는 기능을 합니다.
 위치를 변경하기 전에 어느 방향으로 움직이는지 확인을 하고, 그 방향으로 이동 시 벽인지 검사를 하게 됩니다.
 벽이 아닌 경우 그 방향으로 팩맨의 위치를 이동하고 만약 값이 음가 되거나 배열의 크기를 넘는 경우 새로운 값을 대입하게 됩니다.
 새로운 값을 대입할 수 있는 이유는 사전에 벽인지 검사를 했기 때문입니다.

move_enemies.c :
 적들의 위치를 조정하는 기능을 합니다. 적들은 한 칸씩 움직이거나 제자리에 있게 됩니다.
 게임이 시작된 상태라면 우선 적들의 존재 여부를 확인합니다. 적이 존재하면 사전에 생성한 임의 값을 이용해서 X축으로 이동합니다.
 만약 생성된 임의 값이 0인 경우에는 실질적으로 X축 이동을 하지 않았으므로 Y축 이동을 하게 됩니다.
 [X축 이동확률은 2/3, Y축 이동확률은 2/9, 제자리에 있을 확률은 1/9 입니다.]

ygSonggoon committed:
[add] - [main에 있는 헤더를 묶은 모음집.]

C4NU commented on header.h:
```diff
@@ -0,0 +1,15 @@
+#define RIGHT_MOVE 'd'
+#define PAUSE 'p'
+#define MOVE_STATUS 'm'
+
```
함수별로 .c 파일을 만들며 생각을 해보았는데, 헤더 파트에 구조체도 같이 포함시키는게 덜 혼란스럽지 않을까 하는 생각이 듭니다.

LeftMiddleRight replied:
구조체 선언 부분은 제가 고쳐야 될 부분 같아서 일단 이 선언 기준으로 고쳤습니다
헤더 파일 분리는 아직 안했습니다

ygSonggoon replied:
헤더파일에 구조체를 선언한다는 것을 깜빡했네요. 알려주셔서 고맙습니다. 헤더파일 커밋은 우선 생략해놓겠습니다.

C4NU commented on collision_wall_check.c:
```diff
@@ -4,7 +4,8 @@ int collision_wall_check(Game* g, int i, int j)
     //맵이 배열안에 사용되어있으므로, 배열의 범위를 넘어갈 경우 에러를 알려줍니다.
     if(i < 0 || i > CELL_SIZE-1 || j < 0 || j > CELL_SIZE-1)
-        return 0;
+        printf("OutOfArrayIndex error\n");
```
gotoxy와 같은 특정 좌표에 출력 시키는 함수 없이 바로 printf로 에러 메시지를 출력하게 되면 게임 화면 출력 파트와 충돌이 생기지 않을까요..?

ygSonggoon replied:
우선  멘트 고맙습니다. 제가 C언어라는 것을 깜빡했네요. 우선 해당 오류는 컴파일러가 잡아주긴 하지만, 그러나 제가 굳이 작성한 이유는 나중에 버그나 에러를 잡을때 출력 표시를 해놓으면 도움이 되더라고요. 그래서 굳이 적어놓았습니다.

LeftMiddleRight commented:
이전에 브랜치 한 부분은 리셋했고 아직 삭제는 안 했습니다.
마스터브랜치에서 새로 브랜치해서 작업브랜치로 올렸습니다

ygSonggoon committed:
[fix] - cleancode화 진행(1)

(가독성1) 사전에 약속했던 변수/함수 스타일명을 사용해서 코드 의미를 확실히 했습니다.
(가독성2) 복잡한 조건문을 단순히 했습니다.
(가독성3) 애매하고 복잡한 코드를 확실한 의미를 갖게 했습니다.

[1] move_pacman
(가독성 1)
g->pacman.j++; 무의식적으로 누군가는 i가 X축 j가 Y축이라고 생각합니다.
그러나 해당 코드에서는 i가 Y축을 의미했습니다. 그렇기에 X_axis, Y_axis등 구체적으로 작성했습니다.

(가독성 2)
똑같은 조건문을 매번 사용하며, 보기 안좋게 작성되어 있었습니다. 그렇기에 중복된 조건문은 한번만 사용했고,
switch문을 사용해서 보기좋게 했습니다.

[2] move_enemies
(가독성 3)
적은 (X, Y)좌표를 갖으므로 하나만 검사 혹은 동시에 검사하면 되는데 따로따로 검사를 하고 있었습니다.
또한 X축 이동, Y축이동, 그리고 제자리에 있는 경우를 간단히 나타낼 수 있는데 매우 복잡하게 나타냈습니다.
(원 작성자가 의도했던 확률을 분석해서 그대로 사용했습니다.)
그렇기에 전체적으로 코드를 간단히 이해할 수 있도록 했습니다.

ygSonggoon commented on collision_wall_check.c:
```diff
@@ -4,7 +4,8 @@ int collision_wall_check(Game* g, int i, int j)
     //맵이 배열안에 사용되어있으므로, 배열의 범위를 넘어갈 경우 에러를 알려줍니다.
     if(i < 0 || i > CELL_SIZE-1 || j < 0 || j > CELL_SIZE-1)
-        return 0;
+        printf("OutOfArrayIndex error\n");
```
@ygSonggoon 
cnff
